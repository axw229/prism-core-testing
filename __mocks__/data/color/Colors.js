// @flow
import cloneDeep from 'lodash/cloneDeep'
import sample from 'lodash/sample'

import type { ColorSetPayload, Color } from '../../../src/shared/types/Colors'

export const Colors: ColorSetPayload = {
  "Red": [
    [
      {
        "colorNumber": "6561",
        "coordinatingColors": {
          "coord2ColorId": 1926,
          "coord1ColorId": 2250,
          "whiteColorId": 2681
        },
        "description": [
          "Subdued",
          "Muted",
          "Brilliant",
          "Radiant"
        ],
        "id": "2248",
        "isExterior": true,
        "isInterior": true,
        "name": "Teaberry",
        "rgb": 15454683,
        "lrv": 68.551,
        "colorFamilyNames": [
          "Red"
        ],
        "brandKey": "SW",
        "red": 235,
        "green": 209,
        "blue": 219,
        "hue": 0.935897435897436,
        "saturation": 0.39393939393939387,
        "lightness": 0.8705882352941177,
        "hex": "#ebd1db",
        "isDark": false,
        "cssrgb": "rgb(235,209,219)",
        "storeStripLocator": "102-C1",
        "similarColors": [
          "2255",
          "2262",
          "2269",
          "2662",
          "2535",
          "2276",
          "2658",
          "1974",
          "2543",
          "2531"
        ]
      },
      {
        "colorNumber": "6568",
        "coordinatingColors": {
          "coord2ColorId": 11244,
          "coord1ColorId": 2687,
          "whiteColorId": 2681
        },
        "description": [
          "Subdued",
          "Muted",
          "Brilliant",
          "Radiant"
        ],
        "id": "2255",
        "isExterior": true,
        "isInterior": true,
        "name": "Lighthearted Pink",
        "rgb": 15586781,
        "lrv": 70.758,
        "brandedCollectionIds": [
          31797,
          31791
        ],
        "brandedCollectionNames": [
          "Pottery Barn Fall/Winter 2018",
          "Pottery Barn Fall/Winter 2018"
        ],
        "colorFamilyNames": [
          "Red"
        ],
        "brandKey": "SW",
        "red": 237,
        "green": 213,
        "blue": 221,
        "hue": 0.9444444444444445,
        "saturation": 0.3999999999999999,
        "lightness": 0.8823529411764706,
        "hex": "#edd5dd",
        "isDark": false,
        "cssrgb": "rgb(237,213,221)",
        "storeStripLocator": "103-C1",
        "similarColors": [
          "2248",
          "2262",
          "2269",
          "2276",
          "1974",
          "1981",
          "2658",
          "1995",
          "2662",
          "1988"
        ]
      }
    ]
  ]
}

export function getColors (): ColorSetPayload {
  return cloneDeep(Colors)
}

export function getColor (): Color | void {
  const colors = getColors()
  const fam1 = Object.keys(colors)[0]
  if (colors[fam1]) {
    return sample(colors[fam1][0])
  }
}
