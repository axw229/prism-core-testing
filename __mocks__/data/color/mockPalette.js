const mockPalette = {
  'colors': [
    {
      'colorNumber': '6765',
      'coordinatingColors': {
        'coord2ColorId': '11194',
        'coord1ColorId': '2870',
        'whiteColorId': '2162'
      },
      'description': [
        'Fairly Colorful',
        'Moderately Colorful',
        'Pale'
      ],
      'id': '2452',
      'isExterior': true,
      'isInterior': true,
      'name': 'Spa',
      'lrv': 64.476,
      'brandedCollectionNames': [],
      'colorFamilyNames': [
        'Blue'
      ],
      'brandKey': 'SW',
      'red': 167,
      'green': 220,
      'blue': 220,
      'hue': 0.5,
      'saturation': 0.43089430894308944,
      'lightness': 0.7588235294117647,
      'hex': '#a7dcdc',
      'isDark': false,
      'storeStripLocator': '164-C2',
      'similarColors': [
        '2634',
        '2630',
        '11225',
        '2472',
        '2445',
        '2622',
        '2459',
        '1606',
        '11226',
        '2451'
      ],
      'ignore': false,
      'archived': false
    },
    {
      'colorNumber': '6860',
      'coordinatingColors': {
        'coord2ColorId': '11339',
        'coord1ColorId': '11364',
        'whiteColorId': '2681'
      },
      'description': [
        'Soft',
        'Fairly Bright'
      ],
      'id': '2547',
      'isExterior': true,
      'isInterior': true,
      'name': 'Eros Pink',
      'lrv': 18.874,
      'brandedCollectionNames': [
        '2020 Play',
        '2019 Naturalist'
      ],
      'colorFamilyNames': [
        'Red'
      ],
      'brandKey': 'SW',
      'red': 200,
      'green': 79,
      'blue': 104,
      'hue': 0.9655647382920111,
      'saturation': 0.5238095238095238,
      'lightness': 0.5470588235294118,
      'hex': '#c84f68',
      'isDark': false,
      'storeStripLocator': '105-C5',
      'similarColors': [
        '2273',
        '2266',
        '2542',
        '2546',
        '1998',
        '2280',
        '2530',
        '2287',
        '2279',
        '30193'
      ],
      'ignore': false,
      'archived': false
    },
    {
      'colorNumber': '9018',
      'coordinatingColors': {
        'coord2ColorId': '11218',
        'coord1ColorId': '11200',
        'whiteColorId': '2366'
      },
      'description': [
        'Intense',
        'Pale'
      ],
      'id': '11195',
      'isExterior': false,
      'isInterior': true,
      'name': 'Honey Bees',
      'lrv': 70.14,
      'brandedCollectionNames': [
        '2018 Unity'
      ],
      'colorFamilyNames': [
        'Yellow'
      ],
      'brandKey': 'SW',
      'red': 251,
      'green': 214,
      'blue': 130,
      'hue': 0.11570247933884299,
      'saturation': 0.937984496124031,
      'lightness': 0.7470588235294118,
      'hex': '#fbd682',
      'isDark': false,
      'storeStripLocator': '134-C4',
      'similarColors': [
        '2382',
        '11199',
        '11196',
        '2588',
        '2375',
        '11198',
        '11194',
        '11191',
        '3146',
        '2361'
      ],
      'ignore': false,
      'archived': false
    },
    {
      'colorNumber': '9007',
      'coordinatingColors': {
        'coord2ColorId': '1666',
        'coord1ColorId': '2686',
        'whiteColorId': '2331'
      },
      'description': [
        'Rich',
        'Vibrant',
        'Pale'
      ],
      'id': '11184',
      'isExterior': false,
      'isInterior': true,
      'name': 'Melón Meloso',
      'lrv': 55.314,
      'brandedCollectionNames': [],
      'colorFamilyNames': [
        'Orange'
      ],
      'brandKey': 'SW',
      'red': 242,
      'green': 184,
      'blue': 140,
      'hue': 0.0718954248366013,
      'saturation': 0.7968749999999999,
      'lightness': 0.7490196078431373,
      'hex': '#f2b88c',
      'isDark': false,
      'storeStripLocator': '121-C3',
      'similarColors': [
        '2334',
        '2983',
        '2327',
        '11188',
        '2326',
        '2045',
        '2038',
        '3012',
        '2341',
        '2031'
      ],
      'ignore': false,
      'archived': false
    }
  ],
  'activeColor': {
    'colorNumber': '9007',
    'coordinatingColors': {
      'coord2ColorId': '1666',
      'coord1ColorId': '2686',
      'whiteColorId': '2331'
    },
    'description': [
      'Rich',
      'Vibrant',
      'Pale'
    ],
    'id': '11184',
    'isExterior': false,
    'isInterior': true,
    'name': 'Melón Meloso',
    'lrv': 55.314,
    'brandedCollectionNames': [],
    'colorFamilyNames': [
      'Orange'
    ],
    'brandKey': 'SW',
    'red': 242,
    'green': 184,
    'blue': 140,
    'hue': 0.0718954248366013,
    'saturation': 0.7968749999999999,
    'lightness': 0.7490196078431373,
    'hex': '#f2b88c',
    'isDark': false,
    'storeStripLocator': '121-C3',
    'similarColors': [
      '2334',
      '2983',
      '2327',
      '11188',
      '2326',
      '2045',
      '2038',
      '3012',
      '2341',
      '2031'
    ],
    'ignore': false,
    'archived': false
  },
  'toggleCompareColor': false,
  'temporaryActiveColor': null
}

export default mockPalette
