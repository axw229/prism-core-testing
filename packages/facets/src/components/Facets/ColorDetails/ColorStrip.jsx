// @flow
import React, { useContext } from 'react'
import { useSelector } from 'react-redux'
import filter from 'lodash/filter'
import split from 'lodash/split'
import type { Color, ColorMap } from 'src/shared/types/Colors.js.flow'
import ConfigurationContext, { type ConfigurationContextType } from 'src/contexts/ConfigurationContext/ConfigurationContext'
import * as GA from 'src/analytics/GoogleAnalytics'
import { GA_TRACKER_NAME_BRAND } from 'src/constants/globals'

type Props = { color: Color, onColorChanged: Color => void }
export function ColorStrip ({ color, onColorChanged }: Props) {
  const { brandId }: ConfigurationContextType = useContext(ConfigurationContext)
  const stripLocation = split(color.storeStripLocator, '-')[0]
  const colorMap: ColorMap = useSelector(store => store.colors.items.colorMap)

  // TODO: Perhaps we need to find a better resolution to the data so we aren't having to do this bright test in multiple places
  const stripColors = stripLocation ? filter(colorMap, c => (!c.ignore && split(c.storeStripLocator, '-')[0] === stripLocation)) : []

  return (stripColors.length > 0 &&
    <ul className={`color-info__strip`}>
      <li className={`color-info__strip-location`}>
        <span className={`color-info__strip-location-name`}>{stripLocation}</span>
      </li>
      {stripColors.map((stripColor, i) => {
        const active = color.id === stripColor.id
        return (
          <li key={i} className='color-info__strip-color' style={{ backgroundColor: stripColor.hex }}>
            <button
              className={`color-info__strip-color-info${active ? ` color-info__strip-color-info--active` : ''}`}
              onClick={() => {
                onColorChanged(stripColor)
                GA.event({ category: 'Color Detail / Swatch Chip List', action: 'View Swatch Chip Color', label: stripColor.name }, GA_TRACKER_NAME_BRAND[brandId])
              }}
            >
              <span className={`color-info__strip-color-name${stripColor.isDark ? ` color-info__strip-color-name--dark-color` : ''}${!active ? ' visually-hidden' : ''}`}>
                {stripColor.name}
              </span>
            </button>
          </li>
        )
      })}
    </ul>
  )
}

export default ColorStrip
