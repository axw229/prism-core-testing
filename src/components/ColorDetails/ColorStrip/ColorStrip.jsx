// @flow
import React, { PureComponent } from 'react'
import { connect } from 'react-redux'
import { filter, split } from 'lodash'

import { paintAllSceneSurfaces } from '../../../actions/scenes'

type Props = {
  colors: Array,
  color: Object,
  paintAllSceneSurfaces: Function
}

class ColorStrip extends PureComponent<Props> {
  activeColor: Color

  constructor (props) {
    super(props)

    this.activeColor = props.color
  }

  render () {
    const stripColors = this.colorStripColors()

    return (
      <React.Fragment>
        <strong>Color Strip</strong>
        <hr />
        <ul>
          {stripColors.map(color => {
            return (
              <li key={color.id}>
                <button onClick={() => this.selectColor(color)}>
                  {color.name}{(color.id === this.activeColor.id) ? ' - ACTIVE' : ''}
                </button>
              </li>
            )
          })}
        </ul>
        <hr />
      </React.Fragment>
    )
  }

  colorStripColors () {
    const { colors, color } = this.props
    const stripLocation = split(color.storeStripLocator, '-')[0]

    // grab all colors with the locationId
    return filter(colors, c => split(c.storeStripLocator, '-')[0] === stripLocation)
  }

  selectColor (color) {
    // track the active color within the color strip
    this.activeColor = color

    // paint all the scenes & secen surfaces
    this.props.paintAllSceneSurfaces(color)
  }
}

const mapDispatchToProps = (dispatch: Function) => {
  return {
    paintAllSceneSurfaces: (color) => {
      dispatch(paintAllSceneSurfaces(color))
    }
  }
}

export default connect(null, mapDispatchToProps)(ColorStrip)
