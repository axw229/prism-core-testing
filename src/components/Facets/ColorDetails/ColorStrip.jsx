// @flow
import React from 'react'
import { useSelector } from 'react-redux'
import filter from 'lodash/filter'
import split from 'lodash/split'
import type { Color, ColorMap } from 'src/shared/types/Colors.js.flow'
import * as GA from 'src/analytics/GoogleAnalytics'

type Props = { color: Color, onColorChanged: Color => void }
export function ColorStrip ({ color, onColorChanged }: Props) {
  const stripLocation = split(color.storeStripLocator, '-')[0]
  const colorMap: ColorMap = useSelector(store => store.colors.items.colorMap)

  // TODO: Perhaps we need to find a better resolution to the data so we aren't having to do this bright test in multiple places
  const stripColors = stripLocation ? filter(colorMap, c => (!c.ignore && split(c.storeStripLocator, '-')[0] === stripLocation)) : []

  return (stripColors.length > 0 &&
    <ul className={`color-info__strip`}>
      <li className={`color-info__strip-location`}>
        <span className={`color-info__strip-location-name`}>{stripLocation}</span>
      </li>
      {stripColors.map(stripColor => {
        const active = color.id === stripColor.id
        return (
          <li className='color-info__strip-color' style={{ backgroundColor: stripColor.hex }}>
            <button
              className={`color-info__strip-color-info${active ? ` color-info__strip-color-info--active` : ''}`}
              onClick={() => {
                onColorChanged(stripColor)
                GA.event({ category: 'Color Detail / Swatch Chip List', action: 'View Swatch Chip Color', label: stripColor.name })
              }}
            >
              <span className={`color-info__strip-color-name${stripColor.isDark ? ` color-info__strip-color-name--dark-color` : ''}${!active ? ' visually-hidden' : ''}`}>
                {stripColor.name}
              </span>
            </button>
          </li>
        )
      })}
    </ul>
  )
}

export default ColorStrip
