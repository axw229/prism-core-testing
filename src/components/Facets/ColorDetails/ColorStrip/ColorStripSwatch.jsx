// @flow
import React, { PureComponent } from 'react'
import type { Color } from '../../../../shared/types/Colors'

type Props = {
  color: Color,
  active: Boolean,
  activateColor: Function,
}

class ColorStripSwatch extends PureComponent<Props> {
  static baseClass = 'color-info'

  constructor (props) {
    super(props)

    this.selectColor = this.selectColor.bind(this)
  }

  render () {
    const { color, active } = this.props

    let BUTTON_CLASSES = [
      `${ColorStripSwatch.baseClass}__strip-color-info`
    ]
    if (active) {
      BUTTON_CLASSES.push(`${ColorStripSwatch.baseClass}__strip-color-info--active`)
    }

    return (
      <li className={`${ColorStripSwatch.baseClass}__strip-color`} style={{ backgroundColor: color.hex }}>
        <button className={BUTTON_CLASSES.join(' ')} onClick={this.selectColor}>
          <span className={`${ColorStripSwatch.baseClass}__strip-color-name ${color.isDark ? `${ColorStripSwatch.baseClass}__strip-color-name--dark-color` : ''}`} >{color.name}</span>
        </button>
      </li>
    )
  }

  selectColor = function selectColor () {
    this.props.activateColor(this.props.color)
  }
}

export default ColorStripSwatch
