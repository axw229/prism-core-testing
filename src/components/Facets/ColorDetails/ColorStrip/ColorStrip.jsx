// @flow
import React, { PureComponent } from 'react'
import { filter, split } from 'lodash'

import ColorStripSwatch from './ColorStripSwatch'

type Props = {
  colors: Array,
  color: Object
}

class ColorStrip extends PureComponent<Props> {
  static baseClass = 'color-info'

  state: State = {
    activeColor: void (0)
  }

  constructor (props) {
    super(props)

    this.state = { activeColor: props.color }

    this.activateColor = this.activateColor.bind(this)
  }

  render () {
    const stripColors = this.colorStripColors()

    if (!stripColors) {
      return null
    }

    return (
      <React.Fragment>
        <ul className={`${ColorStrip.baseClass}__strip`}>
          <li className={`${ColorStrip.baseClass}__strip-location`}>
            <span className={`${ColorStrip.baseClass}__strip-location-name`}>{this.colorStripLocation()}</span>
          </li>
          {stripColors.map(color => {
            return <ColorStripSwatch key={color.id} color={color} activateColor={this.activateColor} active={(this.state.activeColor.id === color.id)} />
          })}
        </ul>
      </React.Fragment>
    )
  }

  colorStripLocation () {
    const { color } = this.props
    const stripLocation = split(color.storeStripLocator, '-')[0]

    return stripLocation
  }

  colorStripColors () {
    const { colors } = this.props
    const stripLocation = this.colorStripLocation()

    if (!stripLocation) {
      return null
    }

    // grab all colors with the locationId
    return filter(colors, c => split(c.storeStripLocator, '-')[0] === stripLocation)
  }

  activateColor (color) {
    this.setState({ activeColor: color })
  }
}

export default ColorStrip
