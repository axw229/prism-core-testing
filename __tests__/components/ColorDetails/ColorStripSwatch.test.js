import React from 'react'
import { shallow } from 'enzyme'
import { ColorStripSwatch } from 'src/components/Facets/ColorDetails/ColorStrip/ColorStripSwatch'
import * as Colors from '__mocks__/data/color/Colors'
import ReactGA from 'react-ga'

const color = Colors.getColor()
const active = true
const historyMock = { push: jest.fn() }

const colorStripSwatch = (color) => {
  return shallow(<ColorStripSwatch color={color} history={historyMock} active={active} />)
}

// rendering test
describe('Coordinary Color Swatch component will pass props correctly', () => {
  const wrapper = colorStripSwatch(color)
  it('snapshot testing', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('component will be rendering', () => {
    const colorSwatchExist = wrapper.exists()
    expect(colorSwatchExist).toEqual(true)
  })

  it('component will rendering color swatch data inside of component', () => {
    expect(wrapper.find('span').text()).toEqual(color.name)
  })
})

// event behavior test
describe(('click event test of color swatch component'), () => {
  const wrapper = colorStripSwatch(color)
  it('component should call handclick function', () => {
    const handleClickMock = jest.fn()
    ReactGA.event = handleClickMock
    wrapper.find('button').simulate('click')
    expect(ReactGA.event).toHaveBeenCalled()
    expect(historyMock.push).toHaveBeenCalled()
  })
})
