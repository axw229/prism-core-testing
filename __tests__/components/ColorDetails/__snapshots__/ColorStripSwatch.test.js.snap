// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Coordinary Color Swatch component will pass props correctly snapshot testing 1`] = `
<li
  className="color-info__strip-color"
  style={
    Object {
      "backgroundColor": "#ebd1db",
    }
  }
>
  <button
    className="color-info__strip-color-info color-info__strip-color-info--active"
    onClick={[Function]}
  >
    <span
      className="color-info__strip-color-name "
    >
      Teaberry
    </span>
  </button>
</li>
`;
